drivers/kscan/kscan_mchp_xec_v2.c

```diff
@@ -0,0 +1,454 @@
+/*
```

Comment:
this seems to duplicate significant amount of code with "non-v2", why can't there be a common driver (or at least a shared codebase)?

Reply:
okay, will evaluate

Reply:
This v2 version of driver is applicable to MEC172x device. Although the code is similar, we will be updating the v2 driver later to use pinctrl and there will be device tree specific changes. Also, we want to prioritize on the MEC172x device first.

Reply:
I've just run a diff between the 2 files and honestly, I think a generic driver has to be created, "v2" is duplicating too much code really. "v2" is also misleading.

Reply:
Added common driver instead of v2. 
Please review.

Thanks